### PR TITLE
add FortiPAM client and secret manage API

### PR DESCRIPTION
**Pull request by jerryz1982**

### CODE REVIEW

File: api_client/fortipam_client.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        self._password = password
+        self._key_file = key_file
+        self._cert_file = cert_file
+        self._ca_file = ca_file
+        # SSL server_name_indication
+        self._ssl_sni = ssl_sni
+        self._auto_login = auto_login
+        self._template = templates
+
+    def parse_headers_cookie(self, headers):
```
Comment (samsu) [inline api_client/fortipam_client.py:89]: should be static here

File: api_client/fortipam_client.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+#
+
+from oslo_log import log as logging
+
+from . import exceptions
+from . import base
+from . import constants as const
+from . import client
+from .templates import fortipam as templates
+import re
```
Comment (samsu) [inline api_client/fortipam_client.py:25]: move it to the beginning

File: api_client/templates/fortipam.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                    }
+                ]
+            }
+        ],
+        "group-permission": []
+    }
+}
+"""
+
+# GET_FOLDER = """
```
Comment (samsu) [inline api_client/templates/fortipam.py:182]: remove it

File: api_client/templates/fortipam.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+                "id": 4,
+                "name": "URL",
+                "value": "{{ url | default("") }}"
+            }
+        ]
+    }
+}
+"""
+
+# GET_SECRET = """
```
Comment (samsu) [inline api_client/templates/fortipam.py:110]: remove it

File: api_client/templates/fortipam.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
+LOGIN = """
+{
+    "path": "/XX/YY/ZZ/AUTH",
+    "method": "POST",
+    "body": "magic=175805987&username={{ username }}&password={{ password }}&4Tredir=%2F&4Tmthd=0&domain=local"
+}
+"""
+
+REDIRECT = """
```
Comment (samsu) [inline api_client/templates/fortipam.py:9]: what is it?

**Review by samsu — CHANGES_REQUESTED** [review]

**Comment by samsu** [comment]
please check the comments in the each files.

File: api_client/templates/fortipam.py (hunk 5 above)
Comment (jerryz1982) [reply api_client/templates/fortipam.py:9]: redirect? thats how they do login after login post, here is to follow the redirect

**Review by samsu — APPROVED** [review]

**Review by samsu — APPROVED** [review]

**Review by samsu — APPROVED** [review]

**Pull request merged**